snap/snapcraft.yaml

```diff
@@ -0,0 +1,64 @@
+name: charmcraft
+base: core20
+summary: The charming tool
+license: Apache-2.0
+description: |
+  Charmcraft provides a streamlined, powerful, opinionated, and
+  flexible tool to develop, package, and manage the lifecycle of Juju
+  charm publication, focused particularly on charms written within the
+  Operator Framework.
+
+  It is still in heavy, initial development and so a lot is still To
+  Be Done. However it is already useful, and already simplifies the
+  life of the charmer.
+
+adopt-info: charmcraft  # look for 'snapcraftctl set-*' in the charmcraft part
+architectures: [all]    # nothing platform-dependent here (for now at least)
+
+apps:
+  charmcraft:
+    command: bin/charmcraft
+    plugs:
+      - home            # so it can acess files under the user's home
+      - removable-media # so it can access things in /media, etc. (manually connected)
+      - network         # so 'pip install' can download things
+    environment:
+      # help python pick up useful things from the base snap
+      PYTHONPATH: /usr/lib/python3/dist-packages:$SNAP/lib
+      # have the cache outside of the version dirs (avoids keeping N copies)
+      XDG_CACHE_HOME: $SNAP_USER_COMMON/cache
+
+grade: stable
+confinement: strict
+
+parts:
+  charmcraft:
+    source: .
+    plugin: python
+    # snapcraft uses venv, which doesn't pull in wheel (as opposed to virtualenv)
+    # so then 'pip install PyYAML' gets cross.
+    python-packages: [wheel]
```

Comment:
if you want wheel, don't you also then want Cython for PyYAML? (and pythonX-dev and libyaml-dev, and...)

Reply:
we would, if we were using the yaml that this process builds. But we're not, because core20 already includes it (complete with libyaml), huzzah